jenkinsbuilder/indexdindexymltransform.py

```diff
@@ -0,0 +1,196 @@
+            namespace = ""
+
+            # If a filename contains a ., take everything before the . as project namespace
+            if "." in file_name:
+                namespace = file_name.split(".")[0]
+
+            # Otherwise the project namespace is the entire filename
+            else:
+                namespace = file_name
+
+            potential_index_data = None
+
+            # Read the yaml data in the file
+            with open(potential_index_file, "r") as thefile:
+                potential_index_data = load(thefile)
+
+            # Go through every entry in the yaml data file, extracting what is needed.
+            for entry in potential_index_data:
+                tid = current_index
+                app_id = namespace
+                job_id = entry["job-id"]
+                git_url = entry["git-url"]
+                git_path = entry["git-path"]
+                git_branch = entry["git-branch"]
+                target_file = entry["target-file"]
+                desired_tag = entry["desired-tag"]
+                notify_email = entry["notify-email"]
+                depends_on = entry["depends-on"]
+
+                # Append the collected and generated information to indexyml data collector
+                index_yml["Projects"].append({
+                    "id": tid,
+                    "app-id": app_id,
+                    "job-id": job_id,
+                    "git-url": git_url,
+                    "git-path": git_path,
+                    "git-branch": git_branch,
+                    "target-file": target_file,
+                    "desired-tag": desired_tag,
+                    "notify-email": notify_email,
+                    "depends-on": depends_on
+                })
+
+                current_index += 1
+
+        # Dump the indexyml data to specfied index.yml
+        with open(self._index_yml, "w") as index_yml_file: 
+            safe_dump(index_yml, index_yml_file, default_flow_style=False, indent=2)
+
+        # Add few lines of comment indicating the the file is generated.
+        cl1 = "This file is generated by the CentOS Community Container Pipeline Service"
+        cl2 = "Please do not edit this file manually."
+        commentline = str.format("# {0}\n# {1}\n\n", cl1, cl2)
+
+        with open(self._index_yml, "r") as original:
+            data = original.read()
+
+        with open(self._index_yml, "w") as modified:
+            modified.write(commentline + data)
+
+        return
+
+    def _genindexd(self):
+        """Generate index.d files given index.yml"""
+
+        index_yml = None
+        indexd_data = {}
+
+        # Open the indexyml file and read the yaml data
+        with open(self._index_yml, "r") as index_file:
+            index_yml = load(index_file)
+
+        # For every project in the indexyml file, collect the nessasary data
+        for project in index_yml["Projects"]:
+            app_id = project["app-id"]
```

Comment:
This is for the reverse process, where you have an index.yml but for some reason, want the index.d files.